### ISSUE TITLE
Accept variables in @Requires and requires.filter

### ISSUE THREAD

**Issue by tcalmant** [tcalmant/ipopo#7]
When using a filter in ``@Requires``, it would be interesting to have use a property variable to have an instance-dependent filter.

Something like:
```python
@ComponentFactory()
@Property('_prop_field', 'my.property')
@Requires('_dependency', 'dep_spec', ldap_filter="(svc.value=${my.property})")
...
```

**Comment by tcalmant** [comment]
A new handler will be defined, e.g. ``@RequiresVarFilter``, in order to avoid losses of performance in ``@Requires`` and other existing handlers.

The filter will be parsed when the handler starts and when component properties are updated.
The variables will be injected in the filter string using the [string.Template](https://docs.python.org/3/library/string.html#template-strings) class.

**Comment by tcalmant** [comment]
The first implementation of this handler is live.
The variables are injected using ``str.format`` instead of the ``string.Template`` class.
Some tests have been written for the "single dependency" mode; the "aggregated dependencies" tests will come later.

Note: the implementation isn't atomic: each requirement and each injected service are handled one by one.

**Comment by tcalmant** [comment]
Tests fully implemented after 848b8de79142c05a497fb73dcc99c6e666164f9b: the implementation of RequiresVarFilter can now be considered stable.
